WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderSelectInterestsViewController.swift

```diff
@@ -0,0 +1,112 @@
+    @IBOutlet weak var collectionView: UICollectionView!
+    @IBOutlet weak var buttonContainerView: UIView!
+    @IBOutlet weak var nextButton: FancyButton!
+
+    // MARK: - Mock Data Source
+    private let fakeDataSource = InterestsDataSource(fileName: "interests.json")
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureCollectionView()
+        applyStyles()
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+
+        let layout = collectionView.collectionViewLayout as! ReaderInterestsCollectionViewFlowLayout
+        layout.invalidateLayout()
+    }
+
+    // MARK: - Private Methods
+    private func configureCollectionView() {
+        let nib = UINib(nibName: String(describing: ReaderInterestsCollectionViewCell.self), bundle: nil)
+        collectionView.register(nib, forCellWithReuseIdentifier: Constants.reuseIdentifier)
+
+        guard let layout = collectionView.collectionViewLayout as? ReaderInterestsCollectionViewFlowLayout else {
+            return
+        }
+
+        layout.itemSpacing = Constants.cellSpacing
+        layout.cellHeight = Constants.cellHeight
+    }
+
+    private func applyStyles() {
+        let styleGuide = ReaderInterestsStyleGuide.self
+        styleGuide.applyTitleLabelStyles(label: titleLabel)
+        styleGuide.applySubtitleLabelStyles(label: subTitleLabel)
+        styleGuide.applyNextButtonStyle(button: nextButton)
+
+        buttonContainerView.backgroundColor = ReaderInterestsStyleGuide.buttonContainerViewBackgroundColor
+    }
+}
+
+// MARK: - UICollectionViewDataSource
+extension ReaderSelectInterestsViewController: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return fakeDataSource.count
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell: ReaderInterestsCollectionViewCell = collectionView.dequeueReusableCell(withReuseIdentifier: Constants.reuseIdentifier, for: indexPath) as! ReaderInterestsCollectionViewCell
+        let interest: ReaderInterest = fakeDataSource.interest(for: indexPath.row)
+
+        ReaderInterestsStyleGuide.applyCellLabelStyle(label: cell.label,
+                                                      isSelected: interest.isSelected)
+
+        cell.layer.cornerRadius = Constants.cellCornerRadius
+
+        cell.label.text = interest.title
+
+        return cell
+    }
+}
+
+// MARK: - UICollectionViewDelegate
+extension ReaderSelectInterestsViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        let interest: ReaderInterest = fakeDataSource.interest(for: indexPath.row)
+        interest.isSelected = !interest.isSelected
```

Comment:
When I tap an interest the background and the text color changes in different speeds (first the background, then the text) — and for a brief moment I can see black text in the dark blue background. Do you think this is an easy fix?

Reply:
That's weird, the background color isn't animating and just changing which the text color is animating at the right rate. Not entirely sure why at this point, I just disabled the fade animation for now: https://github.com/wordpress-mobile/WordPress-iOS/pull/14402/commits/bda0ae794a160eeaac858a2a9dfaab2086095aa8